### PR TITLE
Users endpoint

### PR DESCRIPTION
**Pull request by g-krisztian**
ACL fine tuning based on usage

### CODE REVIEW

**Review by ianffcs — APPROVED** [review]

**Comment by dimos-flexiana** [comment]
When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`

```
at clojure.core$name.invokeStatic(core.clj:1595)
at clojure.core$name.invoke(core.clj:1589)
at framework.acl.core$is_allowed.invokeStatic(core.clj:41)
at framework.acl.core$is_allowed.invoke(core.clj:20)
at framework.components.interceptors$acl_restrict$fn__6780.invoke(interceptors.clj:100)
at framework.components.app.core$select_interceptors$fn__8355$fn__8356.invoke(core.clj:73)
at cats.core$bind$fn__865.invoke(core.cljc:96)
at cats.monad.either$reify__1235._mbind(either.cljc:166)
at cats.core$bind.invokeStatic(core.cljc:94)
at cats.core$bind.invoke(core.cljc:81)
at clojure.core.protocols$naive_seq_reduce.invokeStatic(protocols.clj:62)
at clojure.core.protocols$interface_or_naive_reduce.invokeStatic(protocols.clj:72)
at clojure.core.protocols$fn__8159.invokeStatic(protocols.clj:169)
at clojure.core.protocols$fn__8159.invoke(protocols.clj:124)
at clojure.core.protocols$fn__8114$G__8109__8123.invoke(protocols.clj:19)
at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:31)
at clojure.core.protocols$fn__8144.invokeStatic(protocols.clj:75)
at clojure.core.protocols$fn__8144.invoke(protocols.clj:75)
at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13)
at clojure.core$reduce.invokeStatic(core.clj:6828)
at clojure.core$reduce.invoke(core.clj:6810)
at cats.core$_GT__GT__EQ_.invokeStatic(core.cljc:914)
at cats.core$_GT__GT__EQ_.doInvoke(core.cljc:903)
at clojure.lang.RestFn.applyTo(RestFn.java:142)
at clojure.core$apply.invokeStatic(core.clj:665)
at clojure.core$apply.invoke(core.clj:660)
at framework.components.app.core$__GT_app$fn__8423$fn__8425.invoke(core.clj:165)
at ring.adapter.jetty$proxy_handler$fn__8990.invoke(jetty.clj:27)
at ring.adapter.jetty.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
at org.eclipse.jetty.server.Server.handle(Server.java:500)
at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by g-krisztian** [comment]
> When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`
> 
Both the application, and the framework are on users_endpoint branch, and framework is `lein install`-ed?

**Comment by dimos-flexiana** [comment]
> > When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`
> 
> Both the application, and the framework are on users_endpoint branch, and framework is `lein install`-ed?

The **users_endpoint** branch of _Frankie_ depends on the version **0.2.0** of _Framework_ (that has been already released).

*Btw sorry for deleting my first comment. I should have written it on https://github.com/Flexiana/Frankie/pull/85

**Comment by g-krisztian** [comment]
> > > When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`
> > 
> > 
> > Both the application, and the framework are on users_endpoint branch, and framework is `lein install`-ed?
> 
> The **users_endpoint** branch of _Frankie_ depends on the version **0.2.0** of _Framework_ (that has been already released).
> 
> *Btw sorry for deleting my first comment. I should have written it on [Flexiana/Frankie#85](https://github.com/Flexiana/Frankie/pull/85)

The released 0.2.0 is just not working. It does not inject :auth setting from config. That's why I don't preferring this kind of quick version bumping. It fixed in master, but the framework is evolving with Frankie.

**Comment by ianffcs** [comment]
> > > > When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`
> > > 
> > > 
> > > Both the application, and the framework are on users_endpoint branch, and framework is `lein install`-ed?
> > 
> > 
> > The **users_endpoint** branch of _Frankie_ depends on the version **0.2.0** of _Framework_ (that has been already released).
> > *Btw sorry for deleting my first comment. I should have written it on [Flexiana/Frankie#85](https://github.com/Flexiana/Frankie/pull/85)
> 
> The released 0.2.0 is just not working. It does not inject :auth setting from config. That's why I don't preferring this kind of quick version bumping. It fixed in master, but the framework is evolving with Frankie.

It needs to deploy a new version or to remake the tag of the version everytime you merge something to master on framework

File: src/framework/acl/core.clj [hunk 1]
```diff
@@ -17,6 +17,12 @@
   ([uri]
    (re-find #"\w+" uri)))
 
+(defn- user->role
+  [u]
+  (when
+    (or (:users/is_active u) (:is_active u))
```
Comment (LeaveNhA) [inline src/framework/acl/core.clj:23]: Just curiosity, why underscoring?

**Review by LeaveNhA — APPROVED** [review]
Looks good to me.

File: src/framework/acl/core.clj (hunk 1 above)
Comment (g-krisztian) [reply src/framework/acl/core.clj:23]: It's the raw db response, without any transformation. Maybe later we should create a db->model transformer.

**Comment by ianffcs** [comment]
> > > > > > When I start the application in the **users_endpoint** branch and try `curl http://localhost:3000/`, I get `java.lang.NullPointerException`
> > > > > 
> > > > > 
> > > > > Both the application, and the framework are on users_endpoint branch, and framework is `lein install`-ed?
> > > > 
> > > > 
> > > > The **users_endpoint** branch of _Frankie_ depends on the version **0.2.0** of _Framework_ (that has been already released).
> > > > *Btw sorry for deleting my first comment. I should have written it on [Flexiana/Frankie#85](https://github.com/Flexiana/Frankie/pull/85)
> > > 
> > > 
> > > The released 0.2.0 is just not working. It does not inject :auth setting from config. That's why I don't preferring this kind of quick version bumping. It fixed in master, but the framework is evolving with Frankie.
> > 
> > 
> > It needs to deploy a new version or to remake the tag of the version everytime you merge something to master on framework
> 
> Ideally just a well tested version should have a new release.

I think this what differentiates from major and minor releases, but it's a very personal topic

**Pull request merged**